internal/client/consensus/common/block_import.go

```diff
@@ -0,0 +1,169 @@
+type (
+	// Apply precomputed changes coming from block execution or state sync.
+	ApplyChanges struct{}
+)
+
+// Fork choice strategy.
+type ForkChoiceStrategy interface {
+	isForkChoiceStrategy()
+}
+
+type (
+	// Longest chain fork choice.
+	LongestChain struct{}
+	// Custom fork choice rule, where true indicates the new block should be the best block.
+	Custom bool
+)
+
+// Data required to import a Block.
+type BlockImportParams[H runtime.Hash, N runtime.Number] struct {
+	/// Origin of the Block
+	Origin common.BlockOrigin
+	/// The header, without consensus post-digests applied. This should be in the same
+	/// state as it comes out of the runtime.
+	///
+	/// Consensus engines which alter the header (by adding post-runtime digests)
+	/// should strip those off in the initial verification process and pass them
+	/// via the `post_digests` field. During block authorship, they should
+	/// not be pushed to the header directly.
+	///
+	/// The reason for this distinction is so the header can be directly
+	/// re-executed in a runtime that checks digest equivalence -- the
+	/// post-runtime digests are pushed back on after.
+	Header runtime.Header[N, H]
+	/// Justification(s) provided for this block from the outside.
+	Justifications *runtime.Justifications
+	/// Digest items that have been added after the runtime for external
+	/// work, like a consensus signature.
+	PostDigests []runtime.DigestItem
+	/// The body of the block.
+	Body *[]runtime.Extrinsic
+	/// Indexed transaction body of the block.
+	IndexedBody *[][]byte
+	/// Specify how the new state is computed.
+	StateAction StateAction
+	/// Is this block finalized already?
+	Finalized bool
+	/// Intermediate values that are interpreted by block importers. Each block importer,
+	/// upon handling a value, removes it from the intermediate list. The final block importer
+	/// rejects block import if there are still intermediate values that remain unhandled.
+	Intermediates map[string]any
+	/// Auxiliary consensus data produced by the block.
+	/// Contains a list of key-value pairs. If values are `nil`, the keys will be deleted. These
+	/// changes will be applied to `AuxStore` database all as one batch, which is more efficient
+	/// than updating `AuxStore` directly.
+	Auxiliary overlayedchanges.StorageCollection
+	/// Fork choice strategy of this import. This should only be set by a
+	/// synchronous import, otherwise it may race against other imports.
+	/// `nil` indicates that the current verifier or importer cannot yet
+	/// determine the fork choice value, and it expects subsequent importer
+	/// to modify it. If `nil` is passed all the way down to bottom block
+	/// importer, the import fails with an `IncompletePipeline` error.
+	ForkChoice *ForkChoiceStrategy
+	/// Re-validate existing block.
+	ImportExisting bool
+	/// Whether to create "block gap" in case this block doesn't have parent.
+	CreateGap bool
+	/// Cached full header hash (with post-digests applied).
+	PostHash *H
+}
```

Comment:
can you clean up the comments and remove the backticks based on go docs best practices.